spec/support/shared_examples/month_and_year_date_validations.rb

```diff
@@ -0,0 +1,59 @@
+RSpec.shared_examples 'month and year date validations' do |date_field, validations|
+  include DateAndYearConcerns
+
+  let("#{date_field}_day".to_sym) { date.day }
+  let("#{date_field}_month".to_sym) { date.month }
+  let("#{date_field}_year".to_sym) { date.year }
+
+  describe 'when in the future', if: validations[:future] do
+    let(:date) { 40.years.from_now }
+
+    it 'returns :future error' do
+      expect(model).to be_invalid
+
+      expect(model.errors.added?(date_field,
```

Comment:
Nice 👌🏼 . Once https://trello.com/c/ouZ2DExb is in we'll pick up missing keys too, giving us pretty decent test coverage without the verbosity.